python/ray/serve/_private/router.py

```diff
@@ -1001,11 +1006,38 @@ def __init__(
             call_in_event_loop=event_loop,
         )
 
-        self.metrics_pusher = MetricsPusher()
+        # For autoscaling deployments.
         self.autoscaling_config = None
+        # Track queries sent to replicas for the autoscaling algorithm.
+        self.num_queries_sent_to_replicas = defaultdict(int)
+        self._queries_lock = threading.Lock()
```

Comment:
This very much needs a comment especially given the mixture of threading & asyncio model.

Reply:
Added a comment!